source/common/http/filter_manager.h

```diff
@@ -239,6 +239,7 @@ struct ActiveStreamDecoderFilter : public ActiveStreamFilterBase,
   Buffer::BufferMemoryAccountSharedPtr account() const override;
   void setUpstreamOverrideHost(absl::string_view host) override;
   absl::optional<absl::string_view> upstreamOverrideHost() const override;
+  absl::string_view filterConfigName() const override { return filter_context_.config_name; }
```

Comment:
@htuch @wbpcode @adisuissa  [filter_name](https://github.com/envoyproxy/envoy/blob/5e7a4b5f44f398a42897f743a8665ce92fc563a3/envoy/http/filter_factory.h#L35) is mentioned as fallback for config_name. The fallback mechanism is only used in route specific lookup though. So i am wondering is there any edge case `config_name` is empty and we need to fallback here?

Reply:
I was wondering the same and looking for code that does this. We may be better off in doing fallback here as well.

Reply:
Come to think of it, I dont see how this config name can be empty.. the filter format is name + typed config. So there is only one place to get the name.

Reply:
Yea, I was trying to be certain here. 
I agree we are safe (i.e., not probably safe:) ) here after thinking a bit more. 
This must be non-empty string as required by [PGV rule](https://github.com/envoyproxy/envoy/blob/8cfeabd5bd0b15b5e6ef029bf9fef4dcaa1c8254/api/envoy/config/filter/network/http_connection_manager/v2/http_connection_manager.proto#L666). 

Route fallback case is for very specific case to route. It is basically to support the use case that people use  [filter_name](https://github.com/envoyproxy/envoy/blob/8cfeabd5bd0b15b5e6ef029bf9fef4dcaa1c8254/envoy/http/filter_factory.h#L35) rather than [filter_config_name](https://github.com/envoyproxy/envoy/blob/8cfeabd5bd0b15b5e6ef029bf9fef4dcaa1c8254/envoy/http/filter_factory.h#L32) as key of typed_per_filter_config like below.
```
(*route->mutable_typed_per_filter_config())
              [filter_name]
                  .PackFrom(per_route_config);
```